core/backend/image.go

```diff
@@ -0,0 +1,212 @@
+
+	opts := modelOpts(c, o, []model.Option{
+		model.WithBackendString(c.Backend),
+		model.WithAssetDir(o.AssetsDestination),
+		model.WithThreads(uint32(c.Threads)),
+		model.WithContext(o.Context),
+		model.WithModel(c.Model),
+		model.WithLoadGRPCLoadModelOpts(&proto.ModelOptions{
+			CUDA:          c.CUDA || c.Diffusers.CUDA,
+			SchedulerType: c.Diffusers.SchedulerType,
+			PipelineType:  c.Diffusers.PipelineType,
+			CFGScale:      c.Diffusers.CFGScale,
+			LoraAdapter:   c.LoraAdapter,
+			LoraScale:     c.LoraScale,
+			LoraBase:      c.LoraBase,
+			IMG2IMG:       c.Diffusers.IMG2IMG,
+			CLIPModel:     c.Diffusers.ClipModel,
+			CLIPSubfolder: c.Diffusers.ClipSubFolder,
+			CLIPSkip:      int32(c.Diffusers.ClipSkip),
+			ControlNet:    c.Diffusers.ControlNet,
+		}),
+		model.WithExternalBackends(o.ExternalGRPCBackends, false),
+	})
+
+	inferenceModel, err := loader.BackendLoader(
+		opts...,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	fn := func() error {
+		_, err := inferenceModel.GenerateImage(
+			o.Context,
+			&proto.GenerateImageRequest{
+				Height:           int32(height),
+				Width:            int32(width),
+				Mode:             int32(mode),
+				Step:             int32(step),
+				Seed:             int32(seed),
+				CLIPSkip:         int32(c.Diffusers.ClipSkip),
+				PositivePrompt:   positive_prompt,
+				NegativePrompt:   negative_prompt,
+				Dst:              dst,
+				Src:              src,
+				EnableParameters: c.Diffusers.EnableParameters,
+			})
+		return err
+	}
+
+	return fn, nil
+}
+
+func ImageGenerationOpenAIRequest(modelName string, input *datamodel.OpenAIRequest, cl *services.ConfigLoader, ml *model.ModelLoader, startupOptions *datamodel.StartupOptions) (*datamodel.OpenAIResponse, error) {
+	id := uuid.New().String()
+	created := int(time.Now().Unix())
+
+	if modelName == "" {
+		modelName = model.StableDiffusionBackend
+	}
+	log.Debug().Msgf("Loading model: %+v", modelName)
+
+	config, input, err := ReadConfigFromFileAndCombineWithOpenAIRequest(modelName, input, cl, startupOptions)
+	if err != nil {
+		return nil, fmt.Errorf("failed reading parameters from request: %w", err)
+	}
+
+	src := ""
+	if input.File != "" {
+		if strings.HasPrefix(input.File, "http://") || strings.HasPrefix(input.File, "https://") {
+			src, err = utils.CreateTempFileFromUrl(input.File, "", "image-src")
+			if err != nil {
+				return nil, fmt.Errorf("failed downloading file:%w", err)
+			}
+		} else {
+			src, err = utils.CreateTempFileFromBase64(input.File, "", "base64-image-src")
+			if err != nil {
+				return nil, fmt.Errorf("error creating temporary image source file: %w", err)
+			}
+		}
+	}
+
+	log.Debug().Msgf("Parameter Config: %+v", config)
+
+	switch config.Backend {
+	case "stablediffusion":
+		config.Backend = model.StableDiffusionBackend
+	case "tinydream":
+		config.Backend = model.TinyDreamBackend
+	case "":
+		config.Backend = model.StableDiffusionBackend
+	default:
```

Comment:
this is actually a no-op, better drop  default entirely?

Reply:
We had also issues around it, would actually avoid to re-introduce it https://github.com/mudler/LocalAI/pull/1500

Reply:
ah yes - this is actually an issue. If we set a default here we leave the external backends in the cold.

Reply:
```suggestion
```